### ISSUE TITLE
Configuration Methods are Weird

### ISSUE THREAD

**Issue by JKillian** [palantir/tslint#1092]
The different methods for loading configurations in TSLint are having a bit of an internal crisis right now. They work fine and dandy for TSLint, but their names and behaviors are sort of unpredictable, and they're going to have difficulty starting any new relationships with 3rd-party tools until they discover who they are.

```ts
export function loadConfigurationFromPath(configFilePath: string): IConfigurationFile
```
Our above friend here does a fine job of loading configurations. The problem is, it's not really "fromPath" as he'll try to resolve node module locations if you pass a relative path that doesn't start with a ".".

```ts
export function findConfigurationPath(suppliedConfigFilePath: string, inputFilePath: string)
```
`findConfigurationPath` is also somewhat a strange fellow, she takes an optional (but not marked optional) `suppliedConfigFilePath`. If this argument is non-null, this function ensures that the file actually exists and then returns the path as is. Otherwise, it uses `findup` to find a config file given `inputFilePath`.

```ts
export function findConfiguration(configFile: string, inputFilePath: string): IConfigurationFile {
    const configPath = findConfigurationPath(configFile, inputFilePath);
    return loadConfigurationFromPath(configPath);
}
```
Finally, the weirdness all culminates together with this friend of ours. He takes an optional (but not marked optional) param called `configFile` which is passed to `findConfigurationPath`. Now here's where it gets really weird:

Let's say a path like "tslint-config/a-config.json" is passed to `findConfiguration`. `findConfigurationPath` will verify that a file exists at `CWD/tslint-config/a-config.json` and then return the value `tslint-config/a-config.json`. Then, `loadConfigurationFromPath` will receive this value and attempt to resolve it as a node module. Most likely it's not going to exist as a valid file path and a a node module, so an error is going to happen along the way.

This is bad because 3rd-party tools that are using `findConfiguration`, as they should, are going to have issues handling user-provided paths correctly.

---
Possible resolutions:

I think the most logical way to resolve this is to assume that the entry-point configuration is always a local file located via simple paths and never found via node module resolution semantics. This does mean at a bare minimum, a project would need a `tslint.json` file that simply uses `extends` to extend a config file in a node module.

Implementation-wise, `loadConfigurationFromPath` could take an extra optional argument `forceFilePathSemantics=true`. `findConfiguration` and `findConfigurationPath` would then only deal with finding a config file using path semantics, and 3rd-party tools should be able to use `findConfiguration` without incident.

**Comment by JKillian** [comment]
The naming for these methods is still unclear, but I think #1093 should make things less surprising for devs using this API.
